src/rust/ide/enso-protocol/src/file_manager.rs

```diff
@@ -0,0 +1,411 @@
+//! Client library for the JSON-RPC-based File Manager service.
+
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unused_import_braces)]
+#![warn(unused_qualifications)]
+#![warn(unsafe_code)]
+#![warn(missing_copy_implementations)]
+#![warn(missing_debug_implementations)]
+
+use crate::prelude::*;
+
+use crate::common::UTCDateTime;
+
+use json_rpc::api::Result;
+use json_rpc::Handler;
+use json_rpc::make_arg;
+use json_rpc::make_param_map;
+use json_rpc::make_rpc_methods;
+use futures::Stream;
+use serde::Serialize;
+use serde::Deserialize;
+use std::future::Future;
+use uuid::Uuid;
+
+
+
+// =============
+// === Event ===
+// =============
+
+/// Event emitted by the File Manager `Client`.
+pub type Event = json_rpc::handler::Event<Notification>;
+
+
+
+// ============
+// === Path ===
+// ============
+
+/// Path to a file.
+#[derive(Clone,Debug,Display,Eq,Hash,PartialEq,PartialOrd,Ord)]
+#[derive(Serialize, Deserialize)]
+#[derive(Shrinkwrap)]
+pub struct Path(pub String);
```

Comment:
This is in wrong file. Now (in `file_manager.rs`) it is a "file manager path" which is incorrect.

Reply:
But this is "Path used by File Manager Client". Where it should be, if not in module containing File Manager Client?

Reply:
If this is "file path", it should be in `file.rs`. Now it is "file manager path" which is rather not what it should be (It is file path, right?)

Reply:
Right. But not, being in "file_manager" module does not mean that everything is "file-manager", like a modul would be some kind of class.

In ensogl `World` is in module `display`, but that does not mean that this is "world of displays"

We could rename it to FilePath if you really want to.

Reply:
The domain of `Path` is `Enso Protocol`. This type is defined in the protocol, so if anyone is using this crate, it's expected the docs were read.

Reply:
And the protocol actually generalizes file, directory, etc... as `file system objects`. So `Path` is a path to `File System Object`

Reply:
I still don't agree. If we have `display::World` then it is "display's World". Here we have "FileManager's Path" which doesn't make any sense. Either rename it to `FilePath` so you will have `file_manager::FilePath` or move it to `file::Path` and **USE** it here. I understand it is used here, but it should not be defined here. `file_manager::Path` means that it is a path of file manager, not of file system. Like `display::World` means that it is world of display.

Reply:
@wdanilo  Here I disagree.

1)
To me it *is* a path of file manager, not filesystem. 
This is sometthing that file manager API uses, it might have totally nothing in common with filesystem path.

2) Naming should follow the scheme of the enso protocol. Otherwise, it becomes much more difficult to quickoy see what is what and where.

Reply:
@wdanilo And anyway for me it's silly to create a module only to provide a name to one structure. This is not what modules are for. If you want to have extended name o structure, just add it as a name part. 

What is difference if you write `FilePath` instead of `file::Path`?

@notdanilo Michał is right, the Path may be not necessarily a file. So please leave it as Path, and let's stop wasting time for this topic.